Dockerfiles/agent-ot/Dockerfile.byoc

```diff
@@ -0,0 +1,82 @@
+ARG AGENT_VERSION=nightly-ot-beta-main-jmx
+# Use the Ubuntu Slim AMD64 base image
+FROM ubuntu:24.04 AS builder
+
+# Set environment variables
+ARG AGENT_VERSION
+ENV DEBIAN_FRONTEND=noninteractive
+ARG GO_VERSION=1.22.0
+
+# Set the working directory
+WORKDIR /workspace
+
+# Update and install necessary packages
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    curl \
+    ca-certificates \
+    software-properties-common \
+    build-essential \
+    git \
+    python3 \
+    python3-pip \
+    python3-venv \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Go based on architecture
+RUN ARCH=$(dpkg --print-architecture) && \
+    if [ "$ARCH" = "amd64" ]; then \
+    GO_ARCH="linux-amd64"; \
+    elif [ "$ARCH" = "arm64" ]; then \
+    GO_ARCH="linux-arm64"; \
+    else \
+    echo "Unsupported architecture: $ARCH" && exit 1; \
+    fi && \
+    curl -OL https://golang.org/dl/go${GO_VERSION}.$GO_ARCH.tar.gz && \
+    tar -C /usr/local -xzf go${GO_VERSION}.$GO_ARCH.tar.gz && \
+    rm go${GO_VERSION}.$GO_ARCH.tar.gz
+
+# Set up Go environment
+ENV PATH="/usr/local/go/bin:${PATH}"
+RUN mkdir /go
+ENV GOPATH=/go
+
+# Verify installations
+RUN go version && \
+    python3 --version && \
+    curl --version
+
+# Download and unpack source code
+#RUN curl -L https://github.com/DataDog/datadog-agent/archive/refs/tags/${AGENT_VERSION}.tar.gz -o datadog-agent-${AGENT_VERSION}.tar.gz && \
```

Comment:
Why is this commented out? Is this still needed?

Reply:
Once we have an agent release with the otel binary, we should use code of the agent version. We currently use the main branch, which can have flaky or unreleased features/code.

Reply:
OK, I think it makes sense to add comment about it here